lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -112,6 +112,13 @@ sub create_from_settings {
     my ($self, $settings, $scheduled_product_id) = @_;
 
     my %settings = %$settings;
+    foreach my $key (keys %settings) {
+        my $value = $settings{$key};
+
+        if (ref $value eq 'ARRAY' || ref $value eq 'HASH') {
+            $settings{$key} = encode_json($value);
+        }
+    }
```

Comment:
please try if this works

```suggestion
    my %settings = map { ref $settings->{$_} eq 'ARRAY' || ref $settings->{$_} eq 'HASH' ? encode_json($settings->{$_}) : $settings->{$_} } %$settings;
```

or if you want to use a helper variable for the value you could try

```suggestion
    my %settings = map { my $v = $settings->{$_}; ref $v eq 'ARRAY' || ref $v eq 'HASH' ? encode_json($v) : $v } %$settings;
```

Reply:
I would keep using the helper variable. However, I also think the for-loop you wrote initially is fine.

Reply:
we need to think less procedural programming :)

Reply:
getting a `400 Bad Request` for either one of the two suggestions, shouldnt i be modifying `$settings` directly ?

Reply:
yes, I would work on the `%settings` hash, not the `$settings` hashref parameter directly